jadx-plugins/jadx-plugins-api/src/main/java/jadx/api/plugins/utils/ZipSecurity.java

```diff
@@ -73,4 +78,28 @@ public static boolean isValidZipEntry(ZipEntry entry) {
 		return isValidZipEntryName(entry.getName())
 				&& !isZipBomb(entry);
 	}
+
+	public static InputStream getInputStreamForEntry(ZipFile zipFile, ZipEntry entry) throws IOException {
+		InputStream in = zipFile.getInputStream(entry);
+		LimitedInputStream limited = new LimitedInputStream(in, entry.getSize());
+		return new BufferedInputStream(limited);
+	}
+
+	public static void visitZipEntries(File file, BiConsumer<ZipEntry, InputStream> visitor) {
+		try (ZipFile zip = new ZipFile(file)) {
+			zip.stream()
+					.filter(entry -> !entry.isDirectory())
+					.filter(ZipSecurity::isValidZipEntry)
+					.limit(MAX_ENTRIES_COUNT)
+					.forEach(entry -> {
+						try (InputStream in = getInputStreamForEntry(zip, entry)) {
+							visitor.accept(entry, in);
+						} catch (Exception e) {
+							throw new RuntimeException("Error process zip entry: " + entry.getName());
+						}
+					});
+		} catch (Exception e) {
+			throw new RuntimeException("Failed to process zip file: " + file.getAbsolutePath());
```

Comment:
I made a short test with this branch and it seems that this exception at least in the GUI version is never shown to the user. Therefore a user may open a APK file that triggers the ZipSecurity measures but will never know and in the end work with the APK in Jadx that was not fully loaded.

Reply:
You are right, actually, this is a big problem in `jadx-gui`, none of errors shown to the user.
I plan to make some kind of a panel with errors and warnings count if clicked it will open a log window with needed filter.
But showing stacktraces to users is not very good idea, so it is better to also add an additional node in project tree and collect all errors and warnings in a nice report. Anyway, these changes will go in separate commits :)

Reply:
I made a separate issue for this: #986